File: fix.py
import re

with open("src/bot.ts", "r", encoding="utf-8") as f:
    content = f.read()

content = content.replace("Buffer.from(response.data)", "new TextDecoder('utf-8').decode(response)")
content = content.replace(".toString('utf-8')", "")
# Just to be safe, if there's any remaining:
content = content.replace("const content = new TextDecoder('utf-8').decode(response).trim().trim();", "const content = new TextDecoder('utf-8').decode(response).trim();")

with open("src/bot.ts", "w", encoding="utf-8") as f:
    f.write(content)

print("Python fix done again!")

File: fix.js
const fs = require('fs');

let content = fs.readFileSync('src/bot.ts', 'utf8');

// 1. Remove old triggerGithubAction block using regex
content = content.replace(/\/\/ Helper: Trigger GitHub Action[\s\S]*?async function triggerGithubAction.*?catch\s*\(e:\s*any\)\s*\{[\s\S]*?return\s*\{\s*success:\s*false.*?\}\s*;\s*\}\s*\}/, '');

// 2. Fix ArrayBuffer errors using regex
content = content.replace(/const\s+buffer\s*=\s*Buffer\.from\(response\.data\);\s*const\s+content\s*=\s*buffer\.toString\('utf-8'\)\.trim\(\);/g, 'const content = new TextDecoder("utf-8").decode(response).trim();');

content = content.replace(/const\s+buffer\s*=\s*Buffer\.from\(response\.data\);\s*const\s+content\s*=\s*buffer\.toString\('utf-8'\);/g, 'const content = new TextDecoder("utf-8").decode(response);');

fs.writeFileSync('src/bot.ts', content);

let indexContent = fs.readFileSync('src/index.ts', 'utf8');
indexContent = indexContent.replace(/return handleUpdate\(request\);/g, 'return (handleUpdate as any)(request);');
fs.writeFileSync('src/index.ts', indexContent);

console.log("Fixed with Regex");
